source/views/stoprint/printers.js

```diff
@@ -0,0 +1,166 @@
+// @flow
+
+import React from 'react'
+import {SectionList, StyleSheet} from 'react-native'
+import {connect} from 'react-redux'
+import {type ReduxState} from '../../flux'
+import {updatePrinters} from '../../flux/parts/stoprint'
+import type {Printer, PrintJob} from '../../lib/stoprint'
+import {
+	ListRow,
+	ListSeparator,
+	ListSectionHeader,
+	Detail,
+	Title,
+} from '../components/list'
+import type {TopLevelViewPropsType} from '../types'
+import delay from 'delay'
+import toPairs from 'lodash/toPairs'
+import groupBy from 'lodash/groupBy'
+import {StoPrintErrorView} from './components'
+
+const styles = StyleSheet.create({
+	list: {},
+})
+
+type ReactProps = TopLevelViewPropsType & {
+	navigation: {state: {params: {job: PrintJob}}},
+}
+
+type ReduxStateProps = {
+	+printers: Array<Printer>,
+	+recentPrinters: Array<Printer>,
+	+popularPrinters: Array<Printer>,
+	+error: ?string,
+	+loading: boolean,
+	+username: ?string,
+}
+
+type ReduxDispatchProps = {
+	updatePrinters: () => any,
+}
+
+type Props = ReactProps & ReduxDispatchProps & ReduxStateProps
+
+class PrinterListView extends React.PureComponent<Props> {
+	static navigationOptions = {
+		title: 'Select Printer',
+		headerBackTitle: 'Printers',
+	}
+
+	componentDidMount = () => {
+		this.refresh()
+	}
+
+	refresh = async (): any => {
+		let start = Date.now()
+
+		await this.fetchData()
+		// console.log('data returned')
+
+		// wait 0.5 seconds – if we let it go at normal speed, it feels broken.
+		let elapsed = start - Date.now()
+		if (elapsed < 500) {
+			await delay(500 - elapsed)
+		}
+	}
+
+	fetchData = () => this.props.updatePrinters()
+
+	keyExtractor = (item: Printer) => item.printerName
+
+	openPrintRelease = (item: Printer) =>
+		this.props.navigation.navigate('PrintJobReleaseView', {
+			job: this.props.navigation.state.params.job,
+			printer: item,
+			username: this.props.username,
+		})
+
+	renderItem = ({item}: {item: Printer}) => (
+		<ListRow onPress={() => this.openPrintRelease(item)}>
+			<Title>{item.printerName}</Title>
+			<Detail>{item.location}</Detail>
+		</ListRow>
+	)
+
+	renderSectionHeader = ({section: {title}}: any) => (
+		<ListSectionHeader title={title} />
+	)
+
+	render() {
+		if (this.props.error) {
+			return (
+				<StoPrintErrorView
+					navigation={this.props.navigation}
+					refresh={this.refresh}
```

Comment:
Have this call `this.fetchData` instead; there's no need to add the 500ms delay unless you're doing PullToRefresh.